docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx

```diff
@@ -117,48 +124,55 @@ export default function Page() {
 }
 ```
 
-The `useRouter` provides methods such as `push()`, `refresh()`, and more. See the [API reference](/docs/app/api-reference/functions/use-router) for more information.
+For a full list of `useRouter` methods, see the [API reference](/docs/app/api-reference/functions/use-router).
 
 > **Recommendation:** Use the `<Link>` component to navigate between routes unless you have a specific requirement for using `useRouter`.
 
-## How Navigation Works
+## How Routing and Navigation Works
+
+The App Router uses a hybrid approach for routing and navigation. On the server, your application code is automatically code-split by route segments. And on the client, Next.js [prefetches](#1-prefetching) and caches the route segments. This means, when a user navigates to a new route, the browser doesn't reload the page, and only the route segments that change re-render - making navigation between routes instant.
```

Comment:
It's not always instant, but quicker than rerendering the entire page.

Reply:
Thank you!